plugin-server/src/cdp/hog-executor.ts

```diff
@@ -571,6 +574,21 @@ export class HogExecutor {
 
         return values
     }
+
+    private enrichFetchRequest(
+        request: HogFunctionQueueParametersFetchRequest
+    ): HogFunctionQueueParametersFetchRequest {
+        // TRICKY: Some 3rd parties require developer tokens to be passed in the headers
+        // We don't want to expose these to the user so we add them here out of the custom code loop
+
+        request.headers = request.headers ?? {}
+
+        if (request.url.startsWith('https://googleads.googleapis.com/') && !request.headers['developer-token']) {
```

Comment:
We need to hope that there isn't a debugger on https://googleads.googleapis.com/ that exposes the headers..

Reply:
What do you mean by that?

Reply:
There is a risk that Google might offer a webhook.site like debugging tool for requests that might expose the headers including the developer token here.